Matrix.py

```diff
@@ -0,0 +1,175 @@
+""" in this program Matrix is a tuple of tuples and the internal tuples must be as long as the
+external list is.data is the Matrix (tuple) received from the user, and this class has add, sub,
+ mul, scalar_mul and division by scalar methods and non- mathemathic methods
+ as repr, str, hash and eq."""
+
+
+class Matrix:
+    """A matrices calculator for basic math actions
+        Attributes:
+            data(tuple of tuples): a Matrix- each line is a tuple in the major tuple """
+
+    def __init__(self, data):
+        #  inits a Matrix from received data (type= tuple), throws error for non- tuple inputs.
+        check_value = True
+        if type(data) is tuple:
+            for tup in data:
+                if len(tup) != len(data) or type(tup) is not tuple:
+                    check_value *= 0
+        else:
+            check_value *= 0
+        if check_value:
+            self.data = data
+        else:
+            raise ValueError("Invalid matrix")
+
+    def __iter__(self):
+        return iter(self.data)
+
+    def __getitem__(self, item):
+        return self.data[item]
+
+    def __repr__(self):
+        """prints the matrix in a format that can be used for build another Matrix"""
+        return f"Matrix({self.data})"
+
+    # str returns the Matrix in the mathemathic format
```

Comment:
remove, its built-in function, no need for explanation

Reply:
done